src/ContactPeople/ContactPeopleView.js

```diff
@@ -60,172 +66,71 @@ class ContactPeopleView extends React.Component {
-    const addressComplete = get(val, 'addresses', '');
-    const addPhoneNumbers = get(val, 'phoneNumbers', '');
-    const addEmails = get(val, 'emails', '');
-    const addURLS = get(val, 'urls', '');
+        <ContactPersonPhones phones={addPhoneNumbers} />
 
-    return (
-      <div className={css.horizontalLine}>
-        <Row key={key}>
-          {this.printKeyValue('name', fullName, 3, false)}
-          <Col xs={3}>
-            <KeyValue label={<FormattedMessage id="ui-organizations.contactPeople.inactive" />}>
-              <BoolToCheckbox name="Status" value={get(val, ['inactive'])} />
-            </KeyValue>
-          </Col>
-          {this.printKeyValue('categories', language, 3, false)}
-          {this.printKeyValue('language', this.getCategories(val), 3, false)}
-          { addressComplete.length > 0 && (
-            <Fragment>
-              <Col xs={12}>
-                <hr />
-                <div className={css.sub2Headings}>{<FormattedMessage id="ui-organizations.contactPeople.addesses" />}</div>
-              </Col>
-              <Col xs={12}>
-                { addressComplete.map(this.getAddresses) }
-              </Col>
-            </Fragment>
-          )}
-          { addPhoneNumbers.length > 0 && (
-            <Fragment>
-              <Col xs={12}>
-                <hr />
-                <div className={css.sub2Headings}>{<FormattedMessage id="ui-organizations.contactPeople.phoneNumbers" />}</div>
-              </Col>
-              <Col xs={12}>
-                { addPhoneNumbers.map(this.getAddPhoneNumbers) }
-              </Col>
-            </Fragment>
-          )}
-          { addEmails.length > 0 && (
-            <Fragment>
-              <Col xs={12}>
-                <hr />
-                <div className={css.sub2Headings}>{<FormattedMessage id="ui-organizations.contactPeople.emails" />}</div>
-              </Col>
-              <Col xs={12}>
-                { addEmails.map(this.getAddEmails) }
-              </Col>
-            </Fragment>
-          )}
-          { addURLS.length > 0 && (
-            <Fragment>
-              <Col xs={12}>
-                <hr />
-                <div className={css.sub2Headings}>{<FormattedMessage id="ui-organizations.contactPeople.urls" />}</div>
-              </Col>
-              <Col xs={12}>
-                { addURLS.map(this.getAddUrls) }
-              </Col>
-            </Fragment>
-          )}
-        </Row>
+        <ContactPersonEmails emails={addEmails} />
+
+        <ContactPersonURLs urls={addURLS} />
       </div>
     );
   }
 
-  getContacts() {
+  render() {
     const { parentResources } = this.props;
-    const data = ((parentResources || {}).contacts || {}).records || [];
-    if (data.length === 0) return [];
-    return data;
-  }
 
-  render() {
-    if (!isEmpty(this.getContacts())) {
+    if (get(parentResources, 'contacts.isPending', true)) {
+      return <Icon icon="spinner-ellipsis" />;
```

Comment:
Is it a common approach in FOLIO?

Reply:
I think it's common for all applications :) + I need to remount components